clippy_lints/src/pub_underscore_fields.rs

```diff
@@ -37,21 +43,36 @@ declare_clippy_lint! {
 }
 declare_lint_pass!(PubUnderscoreFields => [PUB_UNDERSCORE_FIELDS]);
 
-impl EarlyLintPass for PubUnderscoreFields {
-    fn check_item(&mut self, cx: &EarlyContext<'_>, item: &Item) {
-        // this only pertains to structs
-        let ItemKind::Struct(ref st, _) = item.kind else {
+impl<'tcx> LateLintPass<'tcx> for PubUnderscoreFields {
+    fn check_item(&mut self, cx: &LateContext<'tcx>, item: &'tcx Item<'_>) {
+        // This lint only pertains to structs.
+        let ItemKind::Struct(var, _) = &item.kind else {
```

Comment:
`var` kinda implies "variable", I think the full name `variant_data` would be more illustrative (even more now, that IDEs autocomplete variable names)